packages/dds/tree/src/test/memory/tree.spec.ts

```diff
@@ -95,63 +154,151 @@ describe("SharedTree memory usage", () => {
 				private sharedTree?: TreeView<typeof RootNodeSchema>;
 
 				public async run(): Promise<void> {
 					assert(this.sharedTree?.root.child !== undefined);
 
-					for (let i = 0; i < x; i++) {
-						this.sharedTree.root.child.propertyOne = x;
+					for (let i = 0; i < numberOfEntries; i++) {
+						this.sharedTree.root.child.propertyOne = numberOfEntries;
 					}
 				}
 
 				public beforeIteration(): void {
-					this.sharedTree = createLocalSharedTree("testSharedTree");
+					this.sharedTree = createLocalSharedTree(
+						"testSharedTree",
+						RootNodeSchema,
+						new RootNodeSchema(initialState),
+					);
 				}
 			})(),
 		).timeout(40000); // Set relatively higher threshold as 100_000 iterations can take a while.
 
 		benchmarkMemory(
 			new (class implements IMemoryTestObject {
-				public readonly title = `Set a string property ${x} times in a local SharedTree`;
+				public readonly title =
+					`Set a string property ${numberOfEntries} times in a local SharedTree`;
 				private sharedTree?: TreeView<typeof RootNodeSchema>;
 				public async run(): Promise<void> {
 					assert(this.sharedTree?.root.child !== undefined);
 
-					for (let i = 0; i < x; i++) {
+					for (let i = 0; i < numberOfEntries; i++) {
 						this.sharedTree.root.child.propertyTwo.itemOne = i.toString().padStart(6, "0");
 					}
 				}
 
 				public beforeIteration(): void {
-					this.sharedTree = createLocalSharedTree("testSharedTree");
+					this.sharedTree = createLocalSharedTree(
+						"testSharedTree",
+						RootNodeSchema,
+						new RootNodeSchema(initialState),
+					);
 				}
 			})(),
 		).timeout(40000); // Set relatively higher threshold as 100_000 iterations can take a while.
 
 		benchmarkMemory(
 			new (class implements IMemoryTestObject {
 				public readonly title =
-					`Set an optional integer property ${x} times in a local SharedTree, then clear it`;
+					`Set an optional integer property ${numberOfEntries} times in a local SharedTree, then clear it`;
 				private sharedTree?: TreeView<typeof RootNodeSchema>;
 
 				public async run(): Promise<void> {
 					assert(this.sharedTree?.root.child !== undefined);
 
-					for (let i = 0; i < x; i++) {
-						this.sharedTree.root.child.propertyOne = x;
+					for (let i = 0; i < numberOfEntries; i++) {
+						this.sharedTree.root.child.propertyOne = numberOfEntries;
 					}
 					this.sharedTree.root.child.propertyOne = undefined; // This is possible since the property is optional.
 				}
 
 				public beforeIteration(): void {
-					this.sharedTree = createLocalSharedTree("testSharedTree");
+					this.sharedTree = createLocalSharedTree(
+						"testSharedTree",
+						RootNodeSchema,
+						new RootNodeSchema(initialState),
+					);
 				}
 			})(),
 		).timeout(40000); // Set relatively higher threshold as 100_000 iterations can take a while.
 	}
+
+	function runBenchmarkMemoryForSubTree<TSchema extends ImplicitFieldSchema>(
```

Comment:
This function is a bit complex to read, so it might be nice to give it at least a single sentence doc comment. Maybe something like:

```suggestion
	/**
	 * Define a suite of benchmarks for testing the memory use of variety of sizes of trees of the given schema in various forest implementations.
	  */
	function runBenchmarkMemoryForSubTree<TSchema extends ImplicitFieldSchema>(
```